From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Tue, 1 Mar 2016 15:12:53 +0800
Subject: mmc: dw_mmc-rockchip: fix failing to mount partition with "discard"
Git-commit: c6a9bf99dd458147241ef93817d18e92c3650530
Patch-mainline: v4.7-rc1
References: fate#322003

Without MMC_CAP_ERASE support, we fail to mount partition
with "discard" option since mmc_queue_setup_discard is limited
for checking mmc_can_erase. Without doing mmc_queue_setup_discard,
blk_queue_discard fails to test QUEUE_FLAG_DISCARD flag, so we get
the following log from f2fs(actually similar to other file system):

mounting with "discard" option, but the device does not support discard

Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
Signed-off-by: Jaehoon Chung <jh80.chung@samsung.com>
Signed-off-by: Ulf Hansson <ulf.hansson@linaro.org>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/mmc/host/dw_mmc-rockchip.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/mmc/host/dw_mmc-rockchip.c
+++ b/drivers/mmc/host/dw_mmc-rockchip.c
@@ -241,10 +241,10 @@ static int dw_mci_rockchip_init(struct d
 
 /* Common capabilities of RK3288 SoC */
 static unsigned long dw_mci_rk3288_dwmmc_caps[4] = {
-	MMC_CAP_RUNTIME_RESUME, /* emmc */
-	MMC_CAP_RUNTIME_RESUME, /* sdmmc */
-	MMC_CAP_RUNTIME_RESUME, /* sdio0 */
-	MMC_CAP_RUNTIME_RESUME, /* sdio1 */
+	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE, /* emmc */
+	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE, /* sdmmc */
+	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE, /* sdio0 */
+	MMC_CAP_RUNTIME_RESUME | MMC_CAP_ERASE, /* sdio1 */
 };
 static const struct dw_mci_drv_data rk2928_drv_data = {
 	.prepare_command        = dw_mci_rockchip_prepare_command,
